pages/api/performances/delete.js

```diff
@@ -18,13 +18,42 @@ export default async (req, res) => {
     });
   }
 
-  const deletedAwardPerformance = prisma.awardPerformance.deleteMany({
+  const performance = await prisma.performance.findUnique({
     where: {
-      performance_id: id,
+      id: id,
+    },
+    include: {
+      adjudications: true,
+      awards_performances: true,
     },
   });
 
-  const deletedAdjudications = prisma.adjudication.deleteMany({
+  // If the performance does not exist, we throw an error
+  if (!performance) {
+    return res.status(400).json({
+      error: 'Performance does not exist',
+    });
+  }
+
+  // If the performance has adjudications, we do not allow editing
+  if (performance.adjudications && performance.adjudications.length !== 0) {
+    return res.status(400).json({
+      error: 'Performance cannot be edited as it has an adjudication',
+    });
+  }
+
+  // If the performance is finalized, we do not allow deletion
+  if (performance.awards_performances && performance.awards_performances.length !== 0) {
+    for (const index in performance.awards_performances) {
+      if (performance.awards_performances[index].status === 'FINALIST') {
```

Comment:
`in` is for iterating through object keys. If this is an array, you need to use `of`. Same applies below.

Reply:
Oh oops I mixed it up since the changes I made in awards was iterating through object keys. Fixed.